src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1623 @@
+                    // is declared as successful (ISO/IEC14443-4 card is still in the RF field).
+                    // In case of no or incorrect response, the Result informs about the status of the transaction
+                    // (refer. to §7.1, p:67)
+                    // − Parameter Length : 0,
+                    // − Result Length : 1,
+                    // − Result : 0x00 is OK,
+                    // different from 0x00 is Not OK, Status byte. 
+                    throw new Exception($"Test {diagnoseMode} not implemented");
+                case DiagnoseMode.SelfAntenaTest:
+                    // NumTst = 0x07 : Self Antenna Test
+                    // This test is used to check the continuity of the transmission paths of the antenna.
+                    // − Parameter Length : 1,
+                    // − Parameter : Threshold used for antenna detection
+                    // (applied in register Andet_Control (@610C), see Error!
+                    // Reference source not found.),
+                    // 7 6 5 4 3 2 1 0
+                    // andet_bot andet_up andet_ithl[1:0] andet_ithh[1:0] andet_en
+                    // − Result Length : 1,
+                    // − Result : 0x00 is OK (antenna is detected),
+                    // different from 0x00 is not OK (no antenna is detected). 
+                    throw new Exception($"Test {diagnoseMode} not implemented");
+                default:
+                    break;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Get or set the timeout when PN532 is in virtual card mode
+        /// </summary>
+        public uint VirtualCardTimeout
+        {
+            get { return _virtualCardTimeout * 50; }
+
+            set
+            {
+                // Timeout defines the time-out only in Virtual card configuration (Mode = 0x02).
+                // In Virtual Card mode, this field is mandatory; whereas in the other mode, it is optional.
+                // This parameter indicates the timeout value with a LSB of 50ms.
+                // There is no timeout control if the value is null (Timeout = 0).
+                // The maximum value for the timeout is 12.75 sec (Timeout = 0xFF). 
+                if (value / 50 > 0xFF)
+                    throw new ArgumentException($"{nameof(VirtualCardTimeout)} can't be more than 12750 milliseconds.");
+                _virtualCardTimeout = value / 50;
+                LogInfo.Log($"{nameof(VirtualCardTimeout)} changed: {SetSecurityAccessModule()}", LogLevel.Debug);
+            }
+        }
+
+        /// <summary>
+        /// Get or set the Security Access Module Mode
+        /// </summary>
+        public SecurityAccessModuleMode SecurityAccessModuleMode
+        {
+            get { return _securityAccessModuleMode; }
+            set { LogInfo.Log($"{nameof(SecurityAccessModuleMode)} changed: {SetSecurityAccessModule()}", LogLevel.Debug); }
```

Comment:
this doesn't seem to be doing anything. Could you also split this into multiple lines so that you don't execute anything while formatting?